lib/openstudio-standards/hvac_sizing/Siz.HeatingCoolingFuels.rb

```diff
@@ -196,6 +196,358 @@ def coil_heating_fuels(heating_coil)
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Gas'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Gas_MultiStage'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Water'
+        heat_types << self.coil_heat_type(component)  
+      when 'OS_Coil_Heating_WaterToAirHeatPump_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_WaterToAirHeatPump_VariableSpeed_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_AirToWaterHeatPump'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_Desuperheater'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Node', 'OS_Fan_ConstantVolume', 'OS_Fan_VariableVolume', 'OS_AirLoopHVAC_OutdoorAirSystem'
+        # To avoid extraneous debug messages  
+      else
+        #OpenStudio::logFree(OpenStudio::Debug, 'openstudio.sizing.Model', "No heating fuel types found for #{obj_type}")
+      end
+    end    
+
+    has_electric = false
+    has_fuel = false
+    has_district = false
+    has_solar = false
+    has_hp = false
+    has_mixed = false
+    if heat_types.size > 0
+      heat_types.sort.each do |heat_type|
+         case heat_type
+        when 'Electricity', 'Electric'
+          has_electric = true
+        when 'DistrictHeating', 'District'
+          has_district = true
+        when 'SolarEnergy', 'Solar'
+          has_solar = true
+        when 'NaturalGas', 'Fuel', 'PropaneGas', 'Coal', 'Diesel'
+          has_fuel = true
+        when 'FuelOilNo1', 'FuelOilNo2'
+          has_fuel = true
+        when 'HeatPump'
+          has_hp = true
+        when 'Mixed'
+          has_mixed = true
+        end
+      end
+    end
+
+    # Choose only one heat type, or Mixed if there are multiple
+    heat_type = ''
+    if has_mixed == true
+      heat_type = 'Mixed'
+    end
+    if has_fuel == true && heat_type != 'Fuel'
+      if heat_type == ''
+        heat_type = 'Fuel'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_district == true && heat_type != 'District'
+      if heat_type == ''
+        heat_type = 'District'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_electric == true && heat_type != 'Electric'
+      if heat_type == ''
+        heat_type = 'Electric'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_solar == true && heat_type != 'Solar'
+      if heat_type == ''
+        heat_type = 'Solar'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_hp == true && heat_type != 'HeatPump'
+      # Assume any additional heat types are supplemental
+      heat_type = 'HeatPump'
+    end
+
+    return heat_type
+
+  end
+
+  # Get primary type of heat for one air loop for Appendix G system type selections
+  # 
+  # @param air_loop [object] 
```

Comment:
param air_loop doesn't match argument zone

As written, this method will return the heat type of the last equipment in the loop.  This can be nondeterministic depending on how the model is constructed.  It also precludes the possibility of multiple fuels in zone equipment on the same air loop.

Reply:
@mdahlhausen: I think we need to discuss this. I'm not sure what you mean by "doesn't match argumrent zone". Also, the method builds a list of fuel types, and it returns "Mixed" if more than one is found. This method is only used by our unit tests, not for building the baseline.

Reply:
the method is def zone_equipment_heat_type(zone).  The @param for documentation says air_loop.  It should be 'zone'.

Reply:
also my reading of the code is that the method as written will return the heat_type of the last zone equipment object in `zone.equipment.each do |equipment|`, rather than 'mixed' if there is more than one type in prior equipment.  Let me know if I'm wrong on that.

Reply:
OK, I see your point now, I was looking at the air loop version of the method. It looks like this version was started but not completed, since it was never implemented into the program logic. I think it should be removed, since it is not used and there is already a similar method in the same file: zone_equipment_heating_fuels.